### PR TITLE
 feature/did_resolver_plugin_registry

### PR DESCRIPTION
**Pull request by burdettadam**
(#15 did_resolver_plugin_registry WIP)
Signed-off-by: Adam Burdett <burdettadam@gmail.com>

### CODE REVIEW

File: aries_cloudagent/core/plugin_registry.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def validate_version(self, version_list, module_name):
                 "Versions list must define at least one version module"
             )
 
-        if not all(type(v) is dict for v in version_list):
+        if any(type(v) is not dict for v in version_list):
```
Comment (dbluhm) [inline aries_cloudagent/core/plugin_registry.py:51]: Curious what lead you to make this change

File: aries_cloudagent/core/tests/test_plugin_registry.py [hunk 2]
```diff
@@ -442,19 +442,31 @@ async def test_register_plugin_no_versions(self):
...
                 "str-has-no-versions-attr",  # definition without versions attr
             ]
             assert self.registry.register_plugin("dummy") is None
 
+    async def test_register_plugin_has_setup(self):
+        with async_mock.patch.object(
+            ClassLoader, "load_module", async_mock.MagicMock()
+        ) as load_module:
+            load_module.side_effect = [
+                {"setup": "pressent"},  # module
```
Comment (dbluhm) [inline aries_cloudagent/core/tests/test_plugin_registry.py:457]:
```suggestion
                {"setup": "present"},  # module
```

**Review by dbluhm — COMMENTED** [review]
Looking good so far; just some minor comments/questiosn

File: aries_cloudagent/core/tests/test_plugin_registry.py (hunk 2 above)
Comment (burdettadam) [reply aries_cloudagent/core/tests/test_plugin_registry.py]: good catch! I also realized the duck typing check would not be triggered by a dictionary, so I have changed that code to be a class object instead.

**Review by dbluhm — APPROVED** [review]
LGTM!

**Pull request merged**